lib/string-template.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+/**
+ * Class: StringTemplate
+ *
+ * @param  {String} template The template input
+ * @param  {Object} scope    The scope available for substitution values
+ */
+class StringTemplate {
+  constructor(template, scope) {
+    this.template = template;
+    this.scope = scope;
+  }
+
+  static isTemplate(string) {
+    return this.CAPTURE.test(string);
+  }
+
+  /**
+   * Check if string is a template, and if so, render it.
+   * @param  {String} string Template string
+   * @param  {Object} scope  Parameters available to the template
+   * @return {String}        Rendered string
+   */
+  static coerce(string, scope) {
+    if (!this.isTemplate(string)) {
+      return string;
+    }
+
+    return new this(string, scope).toString();
+  }
+
+  toString() {
+    return this.template.replace(this.constructor.CAPTURE, (match, capture) => {
+      const path = capture.split(this.constructor.DELIMITER);
+      let node = this.scope;
+
+      // Traverse the scope object
+      for (let i = 0; i < path.length; i++) {
+        if (!node.hasOwnProperty(path[i])) {
+          throw new ReferenceError('Undefined parameter ' + capture);
+        }
+        node = node[path[i]];
+      }
+
+      return node;
+    });
+  }
+}
+
+StringTemplate.CAPTURE = /\{\{ ?(.+?) ?\}\}/g;
+StringTemplate.DELIMITER = ':';
+StringTemplate.prototype.toJSON = StringTemplate.prototype.toString;
```

Comment:
Is there a test case for `toJSON`? I didn't see one, but I may have just missed it.